src/couch_db.erl

```diff
@@ -1480,3 +1497,32 @@ normalize_dbname(<<"shards/", _/binary>> = Path) ->
     lists:last(binary:split(mem3:dbname(Path), <<"/">>, [global]));
 normalize_dbname(DbName) ->
     DbName.
+
+validate_dbname(DbName) when is_list(DbName) ->
+    validate_dbname(?l2b(DbName));
+validate_dbname(DbName) when is_binary(DbName) ->
+    Normalized = normalize_dbname(DbName),
+    case couch_db_plugin:validate_dbname(DbName, Normalized) of
+        true ->
+            ok;
+        false ->
+            validate_dbname_int(DbName, Normalized)
+    end.
+
+validate_dbname_int(DbName, Normalized) ->
```

Comment:
an is_binary(DbName) here would be a useful check, it's guaranteed by `validate_dbname` above.